File: examples/02_string_processing.ts
function main(): void {

  const NAME: string = "FerroPhase";

  const VERSION: string = "0.1.0";

  const NAME_LEN: number = 10;

  const VERSION_LEN: number = 5;
  console.log(`name='${NAME}' len=${NAME_LEN}`);
  console.log(`version='${VERSION}' len=${VERSION_LEN}`);

  const IS_EMPTY: boolean = false;

  const IS_LONG: boolean = true;
  console.log(`empty=${IS_EMPTY}, long=${IS_LONG}`);

  const BANNER: string = "FerroPhase v0.1.0";
  console.log(`banner='${BANNER}'`);

  const BUFFER_SIZE: number = 256;
  console.log(`buffer_size=${BUFFER_SIZE}`);
}

main();

File: examples/02_string_processing.rs
pub fn main() -> () {
    pub const NAME: &str = "FerroPhase".to_string();
    pub const VERSION: &str = "0.1.0".to_string();
    pub const NAME_LEN: usize = 10;
    pub const VERSION_LEN: usize = 5;
    println!("name='{}' len={}", NAME, NAME_LEN);
    println!("version='{}' len={}", VERSION, VERSION_LEN);
    pub const IS_EMPTY: bool = false;
    pub const IS_LONG: bool = true;
    println!("empty={}, long={}", IS_EMPTY, IS_LONG);
    pub const BANNER: &str = "FerroPhase v0.1.0".to_string();
    println!("banner='{}'", BANNER);
    pub const BUFFER_SIZE: usize = 256;
    println!("buffer_size={}", BUFFER_SIZE);
}
